### PR TITLE
add WarningInfo. modify Warning struct.

### PR DESCRIPTION
**Pull request by mmou**
for https://github.com/RustSec/cargo-audit/pull/206

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -1,14 +1,11 @@
 //! Error types used by this crate
 
-use chrono;
-use git2;
```
Comment (tarcieri) [inline src/error.rs]: Oh nice, looks like these may have been left over from an edition upgrade...

File: src/report.rs [hunk 2]
```diff
@@ -172,10 +173,28 @@ impl VulnerabilityInfo {
     }
 }
 
+/// Information about warnings
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct WarningInfo {
+    /// Warnings by warning kind
+    pub warnings: HashMap<warning::Kind, Vec<Warning>>,
```
Comment (tarcieri) [inline src/report.rs]: Would suggest using `crate::Map` (imported as `use crate::Map` for consistency (it's an internal re-export of `BTreeMap`, which provides deterministic ordering)

**Comment by mmou** [comment]
should anything be done to update https://github.com/EmbarkStudios/cargo-deny/blob/3121b23bd65f9bed452ce43a7fe40027d493c6f2/src/advisories/mod.rs#L280 before pushing this?

**Comment by tarcieri** [comment]
Heads up @Jake-Shadle, some breaking changes around warnings in the `rustsec` crate

File: src/report.rs [hunk 3]
```diff
@@ -172,10 +173,28 @@ impl VulnerabilityInfo {
...
+}
+
+impl WarningInfo {
+    /// Add warning to WarningInfo
+    pub fn add(&mut self, warning: Warning) {
+        if let Some(w) = self.warnings.get_mut(&warning.kind) {
+            (*w).push(warning);
+        } else {
+            self.warnings.insert(warning.kind.clone(), vec![warning]);
+        }
```
Comment (tarcieri) [inline src/report.rs]: FYI, I'm a fan of using  `BTreeMap::Entry` for this sort of pattern. Here's an example:

https://github.com/RustSec/rustsec-crate/blob/master/src/database/index.rs#L20-L23

Comment (tarcieri) [reply src/report.rs]: And that said, aside from that nit I think this is good to go

**Comment by Jake-Shadle** [comment]
We do consume warnings, but this looks like a fairly trivial change so no worries from my end, appreciate the heads up though @tarcieri!

**Comment by mmou** [comment]
Thanks for the reviews! Lmk if you need anything else from me before you merge.

**Pull request merged**